src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,263 @@
+			&& ES_GRANT_TYPE
+			&& ES_OAUTH_SECRET
+			&& ES_CLIENT_ID
+		);
+	}
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(data) {
+		const contentType = data.query.contentType || 'ALL'; // enum:[FILES,FILES_AND_FOLDERS,COLLECTIONS,ALL]
+		const skipCount = data.query.from || 0;
+		const maxItems = data.query.count || 9;
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = data.query.$ascending || true;
+		const propertyFilter = data.query.propertyFilter || '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || ''; // will give pictures of flowers as default
+
+		// const filterOptions = data.query.filterOptions
+
+
+		if (!this.checkEnv()) {
+			return 'Update your env variables. See --> src/services/edusharing/envTemplate';
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url: `${ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}&skipCount=${skipCount}&maxItems=${maxItems}&sortProperties=${sortProperties}&sortProperties=cm%3Amodified&sortAscending=${sortAscending}&sortAscending=false&propertyFilter=${propertyFilter}&`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] },
+				],
+				facettes: ['cclom:general_keyword'],
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+
+		let eduResponse;
+		try {
+			eduResponse = await request(options);
+		} catch (e) {
+			// eslint-disable-next-line no-console
+			console.error('error: ', e);
+		}
+
+		const parsed = JSON.parse(eduResponse);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
+			mimetype: ['text/html', 'image/jpeg'],
+			provider: ['BauhausMaterial.de', 'München educationcenter', 'Khan Academy'],
+		};
+
+		// filter away everything buy selected mimetype
+		const filterMime = (obj, mimetypes) => {
+			let result = obj.nodes;
+			mimetypes.forEach((type) => { result = result.filter((n) => n.mimeType === type); });
+			return result;
+		};
+
+		// filter away everything buy selected providers
+		const filterProvider = (obj, providers) => obj === providers;
+
+		// adds accesstoken to image-url to let user see the picture on client-side.
+		if (parsed && parsed.nodes) {
+			parsed.nodes.forEach((node) => {
+				if (node.preview && node.preview.url) {
+					node.preview.url += `&accessToken=${this.accessToken}`;
+				}
+			});
+		}
+
+		const filterResult = (obj, opt) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['filterResult' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4977295)